instant.ts

```diff
@@ -74,6 +74,38 @@ async function runTests(path: string) {
   }
 }
 
+const orderPackageIds = (allPackages, chosenPackageIds) => {
+  function resolveDeps(id, currentStack) {
+    if (currentStack.includes(id)) throw Error(`Circular dependency present for id ${id}`)
+    currentStack.push(id)
+
+    if (allPackages[id] && allPackages[id].metadata) {
+      if (
+        !allPackages[id].metadata.dependencies ||
+        !allPackages[id].metadata.dependencies.length
+      ) return [id]
+    } else {
+      throw Error(`Package ${id} does not exist or the metadata is invalid`)
+    }
+
+    let orderedIds = []
+
+    allPackages[id].metadata.dependencies.forEach(dependency => {
+      const ids = resolveDeps(dependency, currentStack)
+      currentStack = [id]
+      orderedIds.push(...[...ids, id])
```

Comment:
I'm seeing an error in testing, I think it's because the `id` needs to only be added after the loop completes.

Reply:
addressed in https://github.com/openhie/instant/pull/63/commits/53bfe709459405ef33b3da3acc088062c7aef375